### PR TITLE
1942 Show Regulatory Authorities to an admin

### PR DESCRIPTION
**Pull request by pezholio**
This builds on #111 to add a view of a regulatory authority on the admin side. The buttons to edit and delete are currently not functional, but will be sorted in th next bit of work. 

# Screenshots 

image

### CODE REVIEW

File: assets/stylesheets/application.scss [hunk 1]
```diff
@@ -3,95 +3,103 @@ $govuk-assets-path: '~govuk-frontend/govuk/assets/';
...
+.rpr-details {
+  &__section-title {
+    text-transform: uppercase;
+  }
 
-.rpr-listing__filter {
-  background-color: govuk-colour('white');
-  height: 180px;
-  overflow-y: auto;
+  &__section-title {
```
Comment (Gweaton) [inline assets/stylesheets/application.scss:16]: Is this a duplicate of the one above, or am I missing something?

File: assets/stylesheets/application.scss [hunk 2]
```diff
@@ -3,95 +3,103 @@ $govuk-assets-path: '~govuk-frontend/govuk/assets/';
...
+  &__ra-summary-list dd {
+    padding: 2px 0;
+  }
 
-.rpr-listing__filter {
-  background-color: govuk-colour('white');
-  height: 180px;
-  overflow-y: auto;
-}
+  &__filters-container {
```
Comment (Gweaton) [inline assets/stylesheets/application.scss:54]: again, is this a duplicate of line 23 here?

File: assets/stylesheets/application.scss [hunk 3]
```diff
@@ -3,95 +3,103 @@ $govuk-assets-path: '~govuk-frontend/govuk/assets/';
...
+    background-color: govuk-colour('light-grey');
+    padding: 1rem;
+  }
 
-.rpr-internal__listing {
-  .govuk-width-container {
-    max-width: 1200px;
+  &__filter {
```
Comment (Gweaton) [inline assets/stylesheets/application.scss:59]: Duplicate of line 27?

File: src/organisations/admin/organisations.controller.spec.ts [hunk 4]
```diff
@@ -31,21 +42,55 @@ jest.mock('../presenters/organisations.presenter', () => {
   };
 });
 
+jest.mock('../presenters/organisation.presenter', () => {
```
Comment (Gweaton) [inline src/organisations/admin/organisations.controller.spec.ts:45]: Interesting, I quite like this approach of mocking out external classes - do you think this gives us enough integration coverage? I suppose this is probably covered in the e2e tests, right?

File: src/organisations/presenters/organisation.presenter.ts [hunk 5]
```diff
@@ -20,7 +20,12 @@ export class OrganisationPresenter {
         html: await this.industries(),
       },
       {
-        text: '',
+        html: `<a class="govuk-link" href="/admin/organisations/${this.organisation.slug}">
+          View details
+          <span class="govuk-visually-hidden">
+            about ${this.organisation.name}
```
Comment (Gweaton) [inline src/organisations/presenters/organisation.presenter.ts:26]: Something I've been thinking about here is, we should probably be escaping all these user-entered values when we render HTML like this. We do it in a lot of places, but I think it'd be good to have a piece of work to tidy it up where relevant.

**Review by Gweaton — APPROVED** [review]
Looks good, a few minor things.

File: assets/stylesheets/application.scss (hunk 3 above)
Comment (pezholio) [reply assets/stylesheets/application.scss:59]: Good catch 👍

File: src/organisations/admin/organisations.controller.spec.ts (hunk 4 above)
Comment (pezholio) [reply src/organisations/admin/organisations.controller.spec.ts:45]: Yeah, it's also coverted by the presenter tests. Trying to be as solitary as possible here 👍

File: src/organisations/presenters/organisation.presenter.ts (hunk 5 above)
Comment (pezholio) [reply src/organisations/presenters/organisation.presenter.ts:26]: That's a good shout. I don't think we get anything for free in the JS standardlib, so I'll add a card to tidy this up later 👍

Comment (Gweaton) [reply src/organisations/presenters/organisation.presenter.ts:26]: Do you have access to view [this PR](https://github.com/ministryofjustice/hmpps-interventions-ui/commit/8995de8b62e3ced7b34fede10833df35df6a84ab)? Where we're interpolating stuff in templates, we can use the `escape` param, but if we're constructing HTML in a presenter, we could use this. I suppose that gives us the benefit of escaping consistently throughout the service.

Comment (pezholio) [reply src/organisations/presenters/organisation.presenter.ts:26]: Ah, nice 👍

**Pull request merged**